src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MongoDbSessionHandler.php

```diff
@@ -69,8 +69,11 @@ class MongoDbSessionHandler implements \SessionHandlerInterface
      */
     public function __construct($mongo, array $options)
     {
-        if (!($mongo instanceof \MongoClient || $mongo instanceof \Mongo)) {
-            throw new \InvalidArgumentException('MongoClient or Mongo instance required');
+        if (!($mongo instanceof \MongoClient || $mongo instanceof \Mongo || $mongo instanceof \MongoDB\Client)) {
+            throw new \InvalidArgumentException('MongoClient, Mongo or MongoDB instance required');
+        }
+        if ($mongo instanceof \MongoDB\Client && !class_exists('\MongoDB\Collection')) {
+            throw new \InvalidArgumentException('\MongoDB\Collection instance not loaded');
```

Comment:
can this really happen? it's possible that a `\MongoDB\Client` is instanciated, but `\MongoDB\Collection` isn't loadable?